src/rust/engine/fs/src/glob_matching.rs

```diff
@@ -399,10 +400,11 @@ trait GlobMatchingImplementation<E: Send + Sync + 'static>: VFS<E> {
   }
 
   fn canonicalize(&self, symbolic_path: PathBuf, link: &Link) -> BoxFuture<Option<PathStat>, E> {
+    let link = link.clone();
```

Comment:
I generally prefer a function to take ownership of a thing, rather than take a reference and clone it, as it both makes the cost more clear, and can avoid the clone if the caller doesn't need the value any more.